### ISSUE TITLE
Error while trying to book rooms in front end

### ISSUE THREAD

**Issue by grog-grog** [tiipiik/OctoberCMS-RoomBooking#15]
Hi! The plugin looks awesome. Unfortunately I am having an issue. When trying to book rooms in the front end, I get the error "The room id field is required". There is no room id field on the form though. Have I missed something or is this a bug? In the back end I have a room set up with a name and a slug.

I noticed also that when I enter a full description for a room in the back end then try to save, I get the error
"Class 'DOMDocument' not found" on line 465 of /var/www/html/vendor/erusev/parsedown-extra/ParsedownExtra.php

Thank you for making a great plugin, I hope I can get it working as this would go down a storm at work :)

**Comment by tiipiik** [comment]
Hi,
I've tested on my local install and I can't reproduce your bugs. Is your OctoberCMS install up to date ?
Can you give me temporaray access to your install so I can make tests ?

**Comment by grog-grog** [comment]
Hi! Thanks for getting back to me.

Unfortunately I'm not allowed to give access, but I am allowed to give you some more info and I can add some screenshots.

Current October build is 419, version of Booking I have is 1.1.12, version of Translate is 1.2.11. The theme I have installed is Flat UI, any modifications to the theme are really basic css changes, nothing major.
![screenshot from 2017-06-30 10-40-42](https://user-images.githubusercontent.com/6556785/27730500-52669116-5d81-11e7-851b-417740d0867c.png)
![screenshot from 2017-06-30 10-41-35](https://user-images.githubusercontent.com/6556785/27730503-526b4652-5d81-11e7-9f4c-b5cd3c8a851f.png)
![screenshot from 2017-06-30 10-41-58](https://user-images.githubusercontent.com/6556785/27730501-526879a4-5d81-11e7-9c6c-9e5115568ab8.png)
![screenshot from 2017-06-30 10-42-53](https://user-images.githubusercontent.com/6556785/27730502-5269677e-5d81-11e7-9b93-cca3cc9e260a.png)

**Comment by tiipiik** [comment]
Did you install other extra plugins ?

Without any access to test and as I can not reproduce yout bugs my help will be extremely limited and I'm sorry for that.

The reason I see could be usage of extra plugins (not core plugins) that would interfere with the RoomBooking plugin. If you have other plugins installed, deactivate them and test again. Also, try to deactivate the Translate plugin and let only the RoomBooking, then try again.

For the id requested, display the source code of the page, search for the room_id field and see if there's a value inside it. But it's possibily due to the bug you have when you create the room. If room is not created properly then maybe the data in the booking page are not rendered properly too.

**Comment by grog-grog** [comment]
Hi, no problem!

All the plugins I have installed are October.Demo, RainLab.Pages, RainLab.Translate and LaminSanneh.FlexiContact. The room creation is unfortunately still bugged if I disable all other plugins. Same if I disable only Translate, or only FlexiContact. I'm not too concerned about this part, it's not a disaster if I don't use this but I thought it might be useful for you to know about.

With the id, I've checked the page source and the room_id I think should be just above the "Book" button. The markup looks like this which I think explains it:

```
<input type="hidden" name="room_id" value="">
    <button type="submit" class="btn btn-primary">Book</button>
</form>
```

**Comment by tiipiik** [comment]
Yes, if room_id is not filed, then you got an error. The question is "why does this fields not filed ?" I'm pretty sure it's related to the error you have on creation. I'll try to make another install of OctoberCMS with the RomBooking plugin to see if I can reproduce your bug.

**Comment by grog-grog** [comment]
That sounds logical. Please let me know if you find anything :)

**Comment by mabasic** [comment]
@grog-grog To solve `"Class 'DOMDocument' not found"` you need to install `php-xml`. In my case that was `php7.2-xml`.

**Comment by tobias-kuendig** [comment]
For people with a missing room id, did you also put the Room component on the same page? Otherwise the `room` property is unavailable to the Booking form component.
